File: AI Lab/08.12.2025(Session1)/grandparent.pl
parent('Hasib' , 'Rakib'). parent('Rakib' , 'Sohel'). parent('Rakib' , 'Rebeka').
parent('Rashid' , 'Hasib'). grandparent(X, Z) :- parent(X, Y), parent(Y, Z).

/* [Built-in KB is enhanced with the 4 facts and 1 rule; two 2-place predicates;   3 variables;   full stop
      (.) as the end marker of a clause/ sentence / statement;   :- as ‘if’;  comma (,) as logical AND. ] */

/* Procedure to find the grandchildren of X */

findGc :- write(' Grandparent: '), read(X), write('Grandchildren: '),
		grandparent(X, Gc), write(Gc), tab(5), fail.
findGc.

File: AI Lab/08.12.2025(Session1)/grandparent.py
tupleList1=[('parent', 'Hasib', 'Rakib'),('parent', 'Rakib', 'Sohel'),
             		('parent', 'Rakib', 'Rebeka'),('parent', 'Rashid', 'Hasib')]
x = str(input("Grandchildren: "))
print("Grandparent: ")
i = 0
while(i<=3):
    if((tupleList1[i][0]=='parent') & (tupleList1[i][2]==x)):
        for j in range(4):
            if((tupleList1[j][0]=='parent') & (tupleList1[i][1]==tupleList1[j][2])):
                print(tupleList1[j][1])
    i=i+1
